### PR TITLE
On inbound transfer precheck page, change unlock check button text after first click.

### PR DESCRIPTION
**Pull request by delputnam**
Start a transfer in using a domain that is transfer locked.
On the pre-check screen, the button text in the first step should read "I've unlocked my domain"
Click the button, and the button text should change to "Check again".

### CODE REVIEW

File: client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx [hunk 1]
```diff
@@ -173,11 +179,13 @@ class TransferDomainPrecheck extends React.Component {
...
 		}
 
-		if ( loading && ! isStepFinished ) {
+		if ( loading && ! isStepFinished && ! unlockCheckClicked ) {
 			message = translate( 'Please wait while we check the lock staus of your domain.' );
 		}
 
-		const buttonText = translate( "I've unlocked my domain" );
+		const buttonText = unlockCheckClicked
+			? translate( 'Check again' )
```
Comment (a8ci18n) [inline client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: Hi! I've found a possible matching string that has already been translated **1** time:
`translate( 'Check Again' )` *ES Score: 14*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx [hunk 2]
```diff
@@ -129,7 +135,7 @@ class TransferDomainPrecheck extends React.Component {
 		} else if ( false === unlocked ) {
 			heading = translate( 'Unlock the domain.' );
 		}
-		if ( loading && ! isStepFinished ) {
+		if ( loading && ! isStepFinished && ! unlockCheckClicked ) {
```
Comment (aidvu) [inline client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: Suppressing these makes them unused anywhere for this section.

File: client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx [hunk 3]
```diff
@@ -173,11 +179,13 @@ class TransferDomainPrecheck extends React.Component {
 			);
 		}
 
-		if ( loading && ! isStepFinished ) {
+		if ( loading && ! isStepFinished && ! unlockCheckClicked ) {
 			message = translate( 'Please wait while we check the lock staus of your domain.' );
```
Comment (aidvu) [inline client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: We have a typo here on `staus`. :(

File: client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx [hunk 4]
```diff
@@ -77,6 +78,11 @@ class TransferDomainPrecheck extends React.Component {
 		this.setState( { currentStep: this.state.currentStep + 1 } );
 	};
 
+	refreshStatus = () => {
+		this.setState( { unlockCheckClicked: true } );
+		this.props.refreshStatus();
```
Comment (aidvu) [inline client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: Might be an overkill, but I'd rather see the button change once the fetching is finished. This way, feels weird. I believe the `refreshStatus()` passed has an argument `callback`, so you can create a function in `precheck` that just does `this.setState( { unlockCheckClicked: true } );` and pass it in there.

File: client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx [hunk 5]
```diff
@@ -173,11 +175,9 @@ class TransferDomainPrecheck extends React.Component {
 			);
 		}
 
-		if ( loading && ! isStepFinished ) {
-			message = translate( 'Please wait while we check the lock staus of your domain.' );
-		}
-
-		const buttonText = translate( "I've unlocked my domain" );
+		const buttonText = unlockCheckClicked
+			? translate( 'Check again' )
```
Comment (a8ci18n) [inline client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: Hi! I've found a possible matching string that has already been translated **1** time:
`translate( 'Check Again' )` *ES Score: 14*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx [hunk 6]
```diff
@@ -77,6 +78,10 @@ class TransferDomainPrecheck extends React.Component {
 		this.setState( { currentStep: this.state.currentStep + 1 } );
 	};
 
+	refreshStatus = () => {
+		this.props.refreshStatus( () => this.setState( { unlockCheckClicked: true } ) );
```
Comment (aidvu) [inline client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: This creates a function each time you call `refreshStatus`. Make it a member of the class and pass it.

File: client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx [hunk 7]
```diff
@@ -119,19 +124,16 @@ class TransferDomainPrecheck extends React.Component {
...
 		const isStepFinished = currentStep > step;
 
 		let heading = translate( "Can't get the domain's lock status." );
 		if ( true === unlocked ) {
 			heading = translate( 'Domain is unlocked.' );
-		} else if ( false === unlocked ) {
+		} else if ( false === unlocked || loading ) {
 			heading = translate( 'Unlock the domain.' );
 		}
-		if ( loading && ! isStepFinished ) {
```
Comment (aidvu) [inline client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: Why completely remove these?

File: client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx [hunk 8]
```diff
@@ -173,11 +175,9 @@ class TransferDomainPrecheck extends React.Component {
 			);
 		}
 
-		if ( loading && ! isStepFinished ) {
```
Comment (aidvu) [inline client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: What I meant is just remove the `! unlockCheckClicked` check. :)

File: client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx (hunk 6 above)
Comment (delputnam) [reply client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: Argh. Yep.  Sorry. :-(

File: client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx (hunk 8 above)
Comment (delputnam) [reply client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: Ah...ok. Sorry...I misunderstood.

**Pull request merged**